### PR TITLE
Improve tooltips

### PR DESCRIPTION
**Pull request by sampletext32**
Improve description of Create SPU Cache tooltip.
Add descriptions to "Never use this" settings.

Offtop.
I've found that lower_spu_thread_priority and accurate_rsx_access do not affect the emulation at all.

### CODE REVIEW

**Comment by AniLeo** [comment]
https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/CPU/CPUThread.cpp#L336

https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/Cell/SPUThread.cpp#L163

**Comment by marcoluc97** [comment]
@sampletext32 maybe future or old features?

- lower_spu_thread_priority `https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/CPU/CPUThread.cpp#L336`

- accurate_rsx_access
`https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/Cell/SPUThread.cpp#L163`

**Comment by AniLeo** [comment]
No, UI setting variables != Config setting variables

File: rpcs3/rpcs3qt/tooltips.h [hunk 1]
```diff
@@ -73,11 +73,11 @@ class Tooltips : public QObject
...
-		const QString ppu_debug                    = tr("Never use this.");
-		const QString spu_debug                    = tr("Never use this.");
-		const QString set_daz_and_ftz              = tr("Never use this.");
-		const QString accurate_getllar             = tr("Never use this.");
-		const QString accurate_putlluc             = tr("Never use this.");
+		const QString ppu_debug                    = tr("Creates PPU logs.\nOnly useful to developers.\nNever use this.");
+		const QString spu_debug                    = tr("Creates SPU logs.\nOnly useful to developers.\nNever use this.");
+		const QString set_daz_and_ftz              = tr("Sets special MXCSR flags to debug errors in SSE operations.\nOnly used in PPU thread when it's not precise.\nOnly useful to developers.\nNever use this.");
+		const QString accurate_getllar             = tr("Accurately processes SPU MFC_GETLLAR operation.\nOnly useful to developers.\nNever use this.");
+		const QString accurate_putlluc             = tr("Accurately processes SPU MFC_PUTLLUC operation.\nOnly useful to developers.\nNever use this.");
```
Comment (elad335) [inline rpcs3/rpcs3qt/tooltips.h:80]: It's not true, those two are not only useful for developers. Keep the first sentence only.

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h:80]: Which one exactly? You've selected 4 of them.

Comment (elad335) [reply rpcs3/rpcs3qt/tooltips.h:80]: last two.

Comment (elad335) [reply rpcs3/rpcs3qt/tooltips.h]: I mean these settings useful sometimes, it was "Never use this" as the developers were too lazy finding the accurate description for it, but yours is good enough for now.

**Comment by sampletext32** [comment]
Fixed a typo in sleep_timers_accuracy
Added description for SPU ASMJIT
Added description for accurate_rsx_access
Specified show_shader_compilation_hint

**Comment by AniLeo** [comment]
Both SPU Recompilers use cache

**Comment by sampletext32** [comment]
> Both SPU Recompilers use cache

Well, in this case I don't fully understand the process. I can rewind this exact change.

File: rpcs3/rpcs3qt/tooltips.h [hunk 2]
```diff
@@ -73,14 +73,14 @@ class Tooltips : public QObject
...
-		const QString ppu_debug                    = tr("Never use this.");
-		const QString spu_debug                    = tr("Never use this.");
-		const QString set_daz_and_ftz              = tr("Never use this.");
-		const QString accurate_getllar             = tr("Never use this.");
-		const QString accurate_putlluc             = tr("Never use this.");
+		const QString ppu_debug                    = tr("Creates PPU logs.\nOnly useful to developers.\nNever use this.");
+		const QString spu_debug                    = tr("Creates SPU logs.\nOnly useful to developers.\nNever use this.");
+		const QString set_daz_and_ftz              = tr("Sets special MXCSR flags to debug errors in SSE operations.\nOnly used in PPU thread when it's not precise.\nOnly useful to developers.\nNever use this.");
+		const QString accurate_getllar             = tr("Accurately processes SPU MFC_GETLLAR operation.\nNever use this.");
+		const QString accurate_putlluc             = tr("Accurately processes SPU MFC_PUTLLUC operation.\nNever use this.");
```
Comment (elad335) [inline rpcs3/rpcs3qt/tooltips.h:80]:
```suggestion
		const QString accurate_putlluc             = tr("Accurately processes SPU MFC_PUTLLUC operation.");
```

Comment (elad335) [reply rpcs3/rpcs3qt/tooltips.h:80]: Same above.

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h]: Done

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h]: Maybe it's worth removing this notification from accurate_rsx_access?

Comment (elad335) [reply rpcs3/rpcs3qt/tooltips.h]: Same with accurate RSX reservations.

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h]: Done

File: rpcs3/rpcs3qt/tooltips.h [hunk 3]
```diff
@@ -60,10 +60,10 @@ class Tooltips : public QObject
 		const QString ppu_llvm                  = tr("Recompiles the game's executable once before running it for the first time.\nThis is by far the fastest option and should always be used.\nShould you face compatibility issues, fall back to one of the Interpreters and retry.\nIf unsure, use this option.");
 		const QString spu_precise               = tr("This is extremely slow but may fix broken graphics in some games.");
 		const QString spu_fast                  = tr("This is slower than the SPU Recompiler but significantly faster than the precise interpreter.\nGames rarely need this however.");
-		const QString spu_asmjit                = tr("This is the fast option with very good compatibility.\nIf unsure, use this option.");
-		const QString spu_llvm                  = tr("This is the fastest option with very good compatibility.\nRecompiles the game's SPU LLVM cache before running which adds extra start-up time.\nIf you experience issues, use the ASMJIT Recompiler.");
+		const QString spu_asmjit                = tr("Recompiles the game's SPU LLVM cache on the go\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
```
Comment (AniLeo) [inline rpcs3/rpcs3qt/tooltips.h:63]: If it uses ASMJIT, it doesn't use LLVM. Both solutions use cache and load them before booting

**Comment by sampletext32** [comment]
I've added some new changes.

File: rpcs3/rpcs3qt/tooltips.h [hunk 4]
```diff
@@ -55,15 +55,15 @@ class Tooltips : public QObject
...
+		const QString spu_precise               = tr("Interprets SPU code with absolute accuracy.\nThis is extremely slow but may fix broken graphics in some games.");
+		const QString spu_fast                  = tr("Interprets SPU code with sacrificed accuracy in order to achieve better performance.\nThis is slower than the SPU Recompiler but significantly faster than the precise interpreter.\nGames rarely need this however.");
+		const QString spu_asmjit                = tr("Recompiles game's SPU code using ASMJIT Recompiler.\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
+		const QString spu_llvm                  = tr("Recompiles and caches game's SPU code using LLVM Recompiler before running which adds extra start-up time.\nThis is the fastest option with very good compatibility.\nIf you experience issues, use the ASMJIT Recompiler.");
+		const QString accurate_xfloat           = tr("Adds extra accuracy to float vectors processing.\nFixes bugs in various games at the cost of performance.\nThis setting is only applied when SPU LLVM is active.");
```
Comment (RainbowCookie32) [inline rpcs3/rpcs3qt/tooltips.h:65]: Accurate xfloat can also be used on the Fast interpreter

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h:65]: Well this piece was there before me. Should I remove this sentence or somehow change it?

Comment (RainbowCookie32) [reply rpcs3/rpcs3qt/tooltips.h:65]: It probably was never updated. Should just change it to reflect that it can be used only on Fast and LLVM.

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h:65]: Ok

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h:65]: I remember some PRs ago some checkboxes were disabled in certain settings set. Maybe not change the tooltip but disable the checkbox at all when spu is not fast or llvm

Comment (RainbowCookie32) [reply rpcs3/rpcs3qt/tooltips.h:65]: The checkbox is already greyed out when other options are selected

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h]: Done

File: rpcs3/rpcs3qt/tooltips.h [hunk 5]
```diff
@@ -55,15 +55,15 @@ class Tooltips : public QObject
...
-		const QString spu_cache                 = tr("Should normally stay enabled.\nDisable this if the cache becomes too large.\nDisabling it does not remove the existing cache.");
+		const QString ppu_precise               = tr("Interprets PPU code with absolute accuracy.\nThis is the most accurate Interpreter, but very slow to play games with.\nYou may try this as a last resort if you encounter odd bugs or crashes.\nIf unsure, use PPU Interpreter Fast or PPU Recompiler (LLVM).");
+		const QString ppu_fast                  = tr("Interprets PPU code with sacrificed accuracy in order to achieve better performance.\nThis is the fastest interpreter.\nIt very rarely breaks games even in comparison to the Precise option.\nTry this if PPU Recompiler (LLVM) fails.");
+		const QString ppu_llvm                  = tr("Recompiles and caches game's PPU code using LLVM Recompiler once before running it for the first time.\nThis is by far the fastest option and should always be used.\nShould you face compatibility issues, fall back to one of the Interpreters and retry.\nIf unsure, use this option.");
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/tooltips.h:60]:
```suggestion
		const QString ppu_llvm                  = tr("Recompiles and caches the game's PPU code using the LLVM Recompiler once before running it for the first time.\nThis is by far the fastest option and should always be used.\nShould you face compatibility issues, fall back to one of the Interpreters and retry.\nIf unsure, use this option.");
```

File: rpcs3/rpcs3qt/tooltips.h [hunk 6]
```diff
@@ -55,15 +55,15 @@ class Tooltips : public QObject
...
+		const QString spu_precise               = tr("Interprets SPU code with absolute accuracy.\nThis is extremely slow but may fix broken graphics in some games.");
+		const QString spu_fast                  = tr("Interprets SPU code with sacrificed accuracy in order to achieve better performance.\nThis is slower than the SPU Recompiler but significantly faster than the precise interpreter.\nGames rarely need this however.");
+		const QString spu_asmjit                = tr("Recompiles game's SPU code using ASMJIT Recompiler.\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/tooltips.h:63]:
```suggestion
		const QString spu_asmjit                = tr("Recompiles the game's SPU code using the ASMJIT Recompiler.\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
```

Comment (sampletext32) [reply rpcs3/rpcs3qt/tooltips.h:63]: Do you think, it's really necessary?

**Pull request merged**